File: cp2/5_week/p1.cpp
#include <bits/stdc++.h>  // This will work only for g++ compiler.
using namespace std;
int tsize;
long tree[5000001];
inline void add(int index, int value) {
	for (int i = index; i < tsize; i += i & (-i)){
		tree[i] += value;
	}

}

inline long query (int index) {
	int ret =0;
	for (int i = index; i > 0; i -= i & (-i))
		ret += tree[i];
	return ret;
}
int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);
	int q;
	cin >> tsize >> q;
	tsize++;
	int index , value , i;
	char operation;
	for (i = 0; i < q; i++) {
		cin >> operation >> index;
		if (operation == '+') {
			cin >> value;
			add (index+1 , value);
		} else {
			cout << query(index) << endl;
		}
	}
	return 0;
}

File: cp2/5_week/p1.py
class Fenwick():
	def __init__(self, size): 
		self.tree = [0]* (size+1)
	def add(self, value, index):
		while index < len(self.tree):
			self.tree[index] += value
			index += index & (-index)
	def query(self, index):
		ret = 0
		while index > 0:
			ret += self.tree[index]
			index -= index & (-index)
		return ret
	def tp(self):
		print(self.tree)

def sol():
	(n,q) = map(int , input().split())
	fen = Fenwick(size = n)
	for i in range(q):
		line = input().split()
		if len(line) == 3:
			(index, value) = map(int , line[1:])
			fen.add(value, index+1)
		else:
			index = int(line[1])
			print(fen.query(index))
sol()
